src/AuthenticatedAPIClient/tests/AuthenticatedAPIClient.test.jsx

```diff
@@ -143,286 +155,239 @@ describe('getAuthenticatedAPIClient', () => {
+    setJwtCookieTo(null);
+    setJwtTokenRefreshResponseTo(200, 'a malformed jwt');
+    expect.hasAssertions();
+    return client.get(mockApiEndpointPath).catch((error) => {
+      // TODO: this error should be truer. Right now the token is not null.
+      expectSingleCallToJwtTokenRefresh();
+      expectNoCallToCsrfTokenFetch();
+      expect(error.message).toEqual('Access token is still null after successful refresh.');
+      expectLogout();
+    });
   });
 
-  describe('ensureAuthenticatedUser', () => {
-    it('redirects to login when no valid JWT', () => {
-      const loginUrl = process.env.LOGIN_URL;
-      const expectedRedirectUrl = encodeURIComponent(process.env.BASE_URL);
-      const expectedLocation = `${loginUrl}?next=${expectedRedirectUrl}`;
-      client.isAccessTokenExpired = jest.fn();
-      client.isAccessTokenExpired.mockReturnValueOnce(true);
-      client.refreshAccessToken = jest.fn();
-      client.refreshAccessToken.mockReturnValueOnce(new Promise((resolve, reject) => {
-        reject({ message: 'Failed!' }); // eslint-disable-line prefer-promise-reject-errors
-      }));
-      client.ensureAuthenticatedUser('');
-      expect(window.location.assign).toHaveBeenCalledWith(expectedLocation);
+  it('logs info for 401 unauthorized api responses', () => {
+    setJwtCookieTo(jwtTokens.valid.encoded);
+    expect.hasAssertions();
+    return client.get('/401').catch(() => {
+      expect(logInfo).toHaveBeenCalledWith('Unauthorized API response from /401');
+      expectRequestToHaveJwtAuth(mock.history.get[0]);
     });
+  });
 
-    it('errors when no valid JWT after coming from login', async () => {
-      jest.spyOn(global.document, 'referrer', 'get').mockReturnValue(process.env.LOGIN_URL);
-      client.isAccessTokenExpired = jest.fn();
-      client.isAccessTokenExpired.mockReturnValueOnce(true);
-      client.refreshAccessToken = jest.fn();
-      client.refreshAccessToken.mockReturnValueOnce(new Promise((resolve, reject) => {
-        reject({ message: 'Failed!' }); // eslint-disable-line prefer-promise-reject-errors
-      }));
-
-      await expect(client.ensureAuthenticatedUser('')).rejects
-        .toThrow(new Error('Redirect from login page. Rejecting to avoid infinite redirect loop.'));
+  it('logs info for 403 forbidden api responses', () => {
+    setJwtCookieTo(jwtTokens.valid.encoded);
+    expect.hasAssertions();
+    return client.get('/403').catch(() => {
+      expect(logInfo).toHaveBeenCalledWith('Forbidden API response from /403');
+      expectRequestToHaveJwtAuth(mock.history.get[0]);
     });
+  });
+});
 
-    it('promise resolves to access token when valid JWT', async () => {
-      mockCookies.get.mockReturnValueOnce(encodedValidJwt);
-      const { authenticatedUser, decodedAccessToken } = await client.ensureAuthenticatedUser('');
+describe('A POST request when the user is logged in ', () => {
```

Comment:
Add one happy path test for the other methods, including OPTIONS, and ensure the right thing is done.